x-pack/plugins/session_view/common/types/process_tree/index.ts

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+export enum EventKind {
+  event = 'event',
+  signal = 'signal',
+}
+
+export enum EventAction {
+  fork = 'fork',
+  exec = 'exec',
+  exit = 'exit',
+  output = 'output',
+}
+
+interface EventActionPartition {
+  fork: ProcessEvent[];
+  exec: ProcessEvent[];
+  exit: ProcessEvent[];
+  output: ProcessEvent[];
+}
+
+interface User {
+  id: string;
+  name: string;
+}
+
+interface ProcessFields {
+  args: string[];
+  args_count: number;
+  command_line: string;
+  entity_id: string;
+  executable: string;
+  interactive: boolean;
+  name: string;
+  working_directory: string;
+  pid: number;
+  pgid: number;
+  user: User;
+  end?: string;
+  exit_code?: number;
+}
+
+export interface ProcessSelf extends ProcessFields {
+  parent: ProcessFields;
+  session: ProcessFields;
+  entry: ProcessFields;
+  last_user_entered?: ProcessFields;
+}
+
+export interface ProcessEvent {
+  '@timestamp': Date;
+  event: {
+    kind: EventKind;
+    category: string;
+    action: EventAction;
+  };
+  host?: {
+    // optional for now (raw agent output doesn't have server identity)
+    architecture: string;
+    hostname: string;
+    id: string;
+    ip: string;
+    mac: string;
+    name: string;
+    os: {
+      family: string;
+      full: string;
+      kernel: string;
+      name: string;
+      platform: string;
+      type: string;
+      version: string;
+    };
+  };
+  process: ProcessSelf;
+  kibana?: {
+    alert: {
+      uuid: string;
+      reason: string;
+      workflow_status: string;
+      status: string;
+      original_time: Date;
+      original_event: {
+        action: string;
+      };
+      rule: {
+        category: string;
+        consumer: string;
+        description: string;
+        enabled: boolean;
+        name: string;
+        query: string;
+        risk_score: number;
+        severity: string;
+        uuid: string;
+      };
+    };
```

Comment:
do you think we might reuse the `host`, `alert`, `rule` types? if we so we should probably extract them into their own types

Reply:
Yea good point. will do